In [1]:
#import classes
from TrandingViewClasses import *

companies_urls = ['NASDAQ-TLSA'] #get from https://www.tradingview.com/symbols/NASDAQ-AAPL/ Support multiple tickers.
companies_data = START.get_data(companies_urls) #Get from DataBase or Website
company = companies_data[0] #get the first company

company_income_statement = IncomeStatementVisualizer(company_data=company)
company_balance_sheet = BalanceSheetVisualizer(company_data=company)
company_cashflow_statement = CashflowStatementVisualizer(company_data=company)
company_statistics_ratios = StatisticsRatiosVisualizer(company_data=company)
company_dividents = DividentsVisualizer(company_data=company)


2023-04-28 20:48:56,426 | INFO | DataBase Connected
2023-04-28 20:48:56,513 | ERROR | ERROR: NASDAQ-TLSA/Income-Statement, not found in tradingview_database.db
2023-04-28 20:48:56,526 | ERROR | ERROR: NASDAQ-TLSA/Balance-Sheet, not found in tradingview_database.db
2023-04-28 20:48:56,542 | ERROR | ERROR: NASDAQ-TLSA/Cashflow-Statement, not found in tradingview_database.db
2023-04-28 20:48:56,547 | ERROR | ERROR: NASDAQ-TLSA/Ratios, not found in tradingview_database.db
2023-04-28 20:48:56,551 | ERROR | ERROR: NASDAQ-TLSA/Company-Data, not found in tradingview_database.db
2023-04-28 20:49:24,616 | INFO | Start balance Sheet Scrape


NoSuchElementException: Message: no such element: Unable to locate element: {"method":"xpath","selector":"//*[@id='FY']"}
  (Session info: headless chrome=87.0.4280.66)
Stacktrace:
Backtrace:
	Ordinal0 [0x009EC0C3+3326147]
	Ordinal0 [0x008D0851+2164817]
	Ordinal0 [0x00757298+619160]
	Ordinal0 [0x006DC070+114800]
	Ordinal0 [0x006FFC42+261186]
	Ordinal0 [0x006F1A2A+203306]
	Ordinal0 [0x006FE559+255321]
	Ordinal0 [0x006F189B+202907]
	Ordinal0 [0x006D3DF4+81396]
	Ordinal0 [0x006D4DEE+85486]
	Ordinal0 [0x006D4D79+85369]
	Ordinal0 [0x008E85DC+2262492]
	GetHandleVerifier [0x00B72874+1487204]
	GetHandleVerifier [0x00B723CD+1486013]
	GetHandleVerifier [0x00B7A368+1518680]
	GetHandleVerifier [0x00B72F4E+1488958]
	Ordinal0 [0x008DED0D+2223373]
	Ordinal0 [0x008EA12B+2269483]
	Ordinal0 [0x008EA26F+2269807]
	Ordinal0 [0x008FECB8+2354360]
	BaseThreadInitThunk [0x76287D49+25]
	RtlInitializeExceptionChain [0x7772B74B+107]
	RtlClearBits [0x7772B6CF+191]


In [12]:
import sqlite3
# Create a connection to the database (this will create the database file if it doesn't exist yet)
conn = sqlite3.connect('tradingview_database.db')
cursor = conn.cursor()

cursor.execute("PRAGMA table_info([NASDAQ-AAPL/Income-Statement])")
columns = cursor.fetchall()
print(columns)
query = f"CREATE TABLE IF NOT EXISTS [NASDAQ-AAPL/Income-Statement] (column1 TEXT, column2 TEXT, column3 TEXT)"
cursor.close()
conn.commit()

[]


In [27]:
import sqlite3

# Connect to the database
conn = sqlite3.connect('tradingview_database.db')
cur = conn.cursor()

# Check if each table has data or is empty
tables = ['NASDAQ-TLSA/Income-Statement', 'NASDAQ-TLSA/Balance-Sheet', 'NASDAQ-TLSA/Cashflow-Statement', 'NASDAQ-TLSA/Ratios', 'NASDAQ-TLSA/Dividents', 'NASDAQ-TLSA/Company-Data']
for table in tables:
    cur.execute(f"SELECT COUNT(*) FROM \"{table}\"")
    rows = cur.fetchone()[0]
    if rows > 0:
        print(f"Table {table} has {rows} rows.")
    else:
        print(f"Table {table} is empty.")

# Disconnect from the database
conn.close()


Table NASDAQ-TLSA/Income-Statement is empty.
Table NASDAQ-TLSA/Balance-Sheet is empty.
Table NASDAQ-TLSA/Cashflow-Statement is empty.
Table NASDAQ-TLSA/Ratios is empty.
Table NASDAQ-TLSA/Dividents is empty.
Table NASDAQ-TLSA/Company-Data has 12 rows.


Dividents

In [ ]:
company_dividents.dividents()

NameError: name 'company_dividents' is not defined

Income Statements

In [ ]:
company_income_statement.revenue()

In [ ]:
company_income_statement.operating_income()

In [ ]:
company_income_statement.pretax_income()

In [ ]:
company_income_statement.discontinued_operations()

In [ ]:
company_income_statement.net_income()

In [ ]:
company_income_statement.diluted_net_income()

In [ ]:
company_income_statement.eps()

In [ ]:
company_income_statement.shares()

In [ ]:
company_income_statement.ebit()

In [ ]:
company_income_statement.operating_expenses()

balance Sheet

In [ ]:
company_balance_sheet.total_assets_liabilities_equity()

In [ ]:
company_balance_sheet.current_non_current_assets()


In [ ]:
company_balance_sheet.current_non_current_liabilities()

In [ ]:
company_balance_sheet.total_debt_net_debt()

In [ ]:
company_balance_sheet.book_value_per_share()

Cashflow Statement

In [ ]:
company_cashflow_statement.cashflow_operating_investing_financial()

In [ ]:
company_cashflow_statement.cashflow_operating_activities()

In [ ]:
company_cashflow_statement.cashflow_investing_activities()

In [ ]:
company_cashflow_statement.cash_from_financing_activities()

Statistics/Ratios

In [ ]:
company_statistics_ratios.shares_outstanding()

In [ ]:
company_statistics_ratios.enterprice_values()

In [ ]:
company_statistics_ratios.numer_of_employees_shareholders()

In [ ]:
company_statistics_ratios.price_ratios()

In [ ]:
company_statistics_ratios.return_ratios()

In [ ]:
company_statistics_ratios.margins()

In [ ]:
company_statistics_ratios.dept_ratios()

In [ ]:
company_statistics_ratios.liquidity_ratios()

In [19]:
from TrandingViewClasses import DataBase

tables = DataBase.ListTableRows('EGX-ORHD/Company-Data')
s = tables[0][0]  # assuming the first element of the first row is the Ticker
Tickerexchange = s.split("-")
Ticker=Tickerexchange[0]
exchange=Tickerexchange[1]
price = tables[7][0]
changes = tables[8][0]
currency = tables[9][0]
print(price)


8.340


[('EGX-ORHD',), ('https://www.tradingview.com/symbols/EGX-ORHD/financials-income-statement/?selected=',), ('https://www.tradingview.com/symbols/EGX-ORHD/financials-balance-sheet/?selected=',), ('https://www.tradingview.com/symbols/EGX-ORHD/financials-cash-flow/?selected=',), ('https://www.tradingview.com/symbols/EGX-ORHD/financials-statistics-and-ratios/?selected=',), ('https://www.tradingview.com/symbols/EGX-ORHD/technicals/',), ('This gauge displays a real-time technical analysis overview for your selected timeframe. The summary of ORASCOM DEVELOPMENT EGYPT is based on the most popular technical indicators, such as Moving Averages, Oscillators and Pivots. Learn more',), ('8.340',), ('+0.12%',), ('EGP',), ('',), ('1',)]

In [3]:
from TrandingViewClasses import *
import yfinance as yf
db=DataBase()
db.Start()
tables=db.GetFromDataBase('EGX-ETEL')
print(tables)
db.Stop()


2023-04-24 14:52:25,035 | INFO | DataBase Connected
2023-04-24 14:52:25,075 | ERROR | ERROR: EGX-ETEL/Income-Statement, not found in tradingview_database.db
2023-04-24 14:52:25,089 | ERROR | ERROR: EGX-ETEL/Balance-Sheet, not found in tradingview_database.db
2023-04-24 14:52:25,104 | ERROR | ERROR: EGX-ETEL/Cashflow-Statement, not found in tradingview_database.db
2023-04-24 14:52:25,109 | ERROR | ERROR: EGX-ETEL/Ratios, not found in tradingview_database.db
2023-04-24 14:52:25,113 | ERROR | ERROR: EGX-ETEL/Company-Data, not found in tradingview_database.db
None
2023-04-24 14:52:25,118 | INFO | DataBase Disconnected
